source/reference/minio-server/settings/console.rst

```diff
@@ -176,6 +178,212 @@ Log Query URL
       This setting does not have a configuration variable setting.
       Use the Environment Variable instead.
 
+Content Security Policy
+~~~~~~~~~~~~~~~~~~~~~~~
+
+*Optional*
+
+Configure MinIO Console to generate a `Content-Security-Policy <https://en.wikipedia.org/wiki/Content_Security_Policy>`__ header in HTTP responses.
+Defaults to ``default-src 'self' 'unsafe-eval' 'unsafe-inline';``
+
+.. tab-set::
+
+   .. tab-item:: Environment Variable
+      :sync: envvar
+
+      .. envvar:: MINIO_BROWSER_CONTENT_SECURITY_POLICY
+
+         .. code-block:: shell
+            :class: copyable
+
+            set MINIO_BROWSER_CONTENT_SECURITY_POLICY="default-src 'self' 'unsafe-eval' 'unsafe-inline';"
+
+   .. tab-item:: Configuration Setting
+      :sync: config
+
+      .. mc-conf:: browser csp_policy
+         :delimiter: " "
+
+         .. code-block:: shell
+            :class: copyable
+		 
+            mc admin config set browser \
+               csp_policy="default-src 'self' 'unsafe-eval' 'unsafe-inline';" \
+               [ARGUMENT=VALUE ...]
+
+
+Strict Transport Security
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+*Optional*
+
+Configure MinIO console to generate a `Strict-Transport-Security <https://en.wikipedia.org/wiki/HTTP_Strict_Transport_Security>`__ header in HTTP responses.
+
+To generate the header, you **must** set a duration using either :envvar:`MINIO_BROWSER_HSTS_SECONDS` or :mc-conf:`~browser.hsts_seconds`.
+Other HSTS settings are optional.
+
+.. tab-set::
+
+   .. tab-item:: Environment Variables
+      :sync: envvar
+
+      .. envvar:: MINIO_BROWSER_HSTS_INCLUDE_SUB_DOMAINS
+
+         Set to ``on`` to also apply the configured HSTS policy to all MinIO Console subdomains.
+         Defaults to ``off``.
+
+         .. code-block:: shell
+            :class: copyable
+
+            set MINIO_BROWSER_HSTS_INCLUDE_SUB_DOMAINS="on"
+
+      .. envvar:: MINIO_BROWSER_HSTS_PRELOAD
+
+         Set to ``on`` to direct the client browser to add the MinIO Console domain to its HSTS preload list.
+         Defaults to ``off``.
+
+         .. code-block:: shell
+            :class: copyable
+
+            set MINIO_BROWSER_HSTS_INCLUDE_SUB_DOMAINS="on"
+
+      .. envvar:: MINIO_BROWSER_HSTS_SECONDS
```

Comment:
Consider making this the first one, since it's the one required to generate the header.